pytorch_lightning/loops/fit_loop.py

```diff
@@ -234,3 +233,8 @@ def should_accumulate(self) -> bool:
 
     def teardown(self) -> None:
         self.epoch_loop.teardown()
+
+    def _prepare_dataloader_iter(self) -> Iterator:
+        dataloader = self.trainer.accelerator.process_dataloader(self.trainer.train_dataloader)
+        dataloader = self.trainer.data_connector.get_profiled_dataloader(dataloader)
+        return iter(dataloader)
```

Comment:
can we have this as a utility function outside? Currently this is duplicated in evaluation_loop

Reply:
Removed directly !